main/src/ca/uwaterloo/flix/language/phase/unification/QuineMcCluskey.scala

```diff
@@ -0,0 +1,363 @@
+/*
+ *  Copyright 2022 Anna Blume Jakobsen
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package ca.uwaterloo.flix.language.phase.unification
+
+import ca.uwaterloo.flix.language.ast.{SourceLocation, Symbol, Type}
+import ca.uwaterloo.flix.util.InternalCompilerException
+import ca.uwaterloo.flix.util.collection.Bimap
+
+import scala.collection.mutable
+import scala.collection.immutable.IntMap
+
+/**
+  * The representation of a variable being true, false
+  * or either ("don't care").
+  */
+sealed trait BoolVal
+
+object BoolVal {
+  case object True extends BoolVal
+
+  case object False extends BoolVal
+
+  case object DontCare extends BoolVal
+}
+
+object QuineMcCluskey {
+  /**
+    * A global object to run the Quine-McCluskey algorithm
+    */
+  val Global: QuineMcCluskey = new QuineMcCluskey()
+}
+
+class QuineMcCluskey {
```

Comment:
You can just declare QuineMcCluskey as an object, you dont need an object and then a global instance.